File: examples/features/standard/rest/dup-send/src/main/java/ReceiveOrder.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License. You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import org.jboss.resteasy.client.ClientRequest;
import org.jboss.resteasy.client.ClientResponse;
import org.jboss.resteasy.spi.Link;

public class ReceiveOrder {

   public static void main(String[] args) throws Exception {
      // first get the create URL for the shipping queue
      ClientRequest request = new ClientRequest("http://localhost:8080/queues/orders");
      ClientResponse res = request.head();
      Link pullConsumers = res.getHeaderAsLink("msg-pull-consumers");
      res.releaseConnection();
      res = pullConsumers.request().post();
      Link consumeNext = res.getHeaderAsLink("msg-consume-next");
      res.releaseConnection();
      while (true) {
         System.out.println("Waiting...");
         res = consumeNext.request().header("Accept-Wait", "10").post();
         if (res.getStatus() == 503) {
            System.out.println("Timeout...");
            consumeNext = res.getHeaderAsLink("msg-consume-next");
         } else if (res.getStatus() == 200) {
            Order order = (Order) res.getEntity(Order.class);
            System.out.println(order);
            consumeNext = res.getHeaderAsLink("msg-consume-next");
         } else {
            throw new RuntimeException("Failure! " + res.getStatus());
         }
         res.releaseConnection();
      }
   }
}

File: examples/features/standard/rest/jms-to-rest/receiveOrder.py
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.
import httplib
import urlparse

conn = httplib.HTTPConnection("localhost:8080")
conn.request("HEAD", "/queues/orders")
res = conn.getresponse()
consumersLink = res.getheader("msg-pull-consumers")
consumersParsed = urlparse.urlparse(consumersLink)
conn = httplib.HTTPConnection(consumersParsed.netloc)
conn.request("POST", consumersParsed.path)
res = conn.getresponse()
consumeLink = res.getheader("msg-consume-next")
session = res.getheader("Location")
print consumeLink
conn.close()

headers = {"Accept-Wait": "3", "Accept": "application/xml"}

try:
    print "Waiting..."
    while True:
        createParsed = urlparse.urlparse(consumeLink)
        conn = httplib.HTTPConnection(createParsed.netloc)
        conn.request("POST", createParsed.path, None, headers)
        res = conn.getresponse()
        if res.status == 503:
            consumeLink = res.getheader("msg-consume-next")
        elif res.status == 200:
            print "Success!"
            data = res.read()
            print data
            consumeLink = res.getheader("msg-consume-next")
            print "Waiting"
        else:
            raise Exception('failed')
finally:
    if session is not None:
        print "deleting activemq session..."
        createParsed = urlparse.urlparse(session)
        conn = httplib.HTTPConnection(createParsed.netloc)
        conn.request("DELETE", createParsed.path)
        res = conn.getresponse()











